datanode/sqlstore/accounts.go

```diff
@@ -64,13 +64,6 @@ func (as *Accounts) Add(ctx context.Context, a *entities.Account) error {
 }
 
 func (as *Accounts) GetByID(ctx context.Context, accountID entities.AccountID) (entities.Account, error) {
-	if account, ok := as.getAccountFromCache(accountID); ok {
-		return account, nil
-	}
-
-	as.cacheLock.Lock()
-	defer as.cacheLock.Unlock()
-
```

Comment:
This cache is pretty critical - without it we are constantly hammering the DB for account IDs; though perhaps there's some more cunningness in the PR I haven't read yet that makes us not need it

Reply:
As in the code still tries to read from the cache later no.